test/test_sparse.py

```diff
@@ -3219,14 +3216,13 @@ def sparse_log(x):
     # TODO: Check after why ROCm's cusparseXcsrgemm2Nnz function doesn't return the same nnz value as CUDA
     @skipIfRocm
     @coalescedonoff
-    @dtypes(*get_all_complex_dtypes(),
-            *get_all_fp_dtypes(include_half=False, include_bfloat16=False))
-    @dtypesIfCUDA(*((torch.complex64,) if CUDA11OrLater else ()),
-                  *((torch.complex128,) if CUSPARSE_SPMM_COMPLEX128_SUPPORTED else ()),
-                  *get_all_fp_dtypes(
-                      include_half=(CUDA11OrLater and SM53OrLater),
-                      include_bfloat16=(CUDA11OrLater and SM80OrLater)))
-    @precisionOverride({torch.bfloat16: 2.5e-2, torch.float16: 2.5e-2, torch.complex64: 1e-2, torch.float32: 1e-2})
+    @dtypes(*floating_and_complex_types())
+    @dtypesIfCUDA(*[torch.complex64] if CUDA11OrLater else [],
+                  *[torch.complex128] if CUSPARSE_SPMM_COMPLEX128_SUPPORTED else [],
+                  *floating_types_and(
+                      *[torch.half] if CUDA11OrLater and SM53OrLater else [],
+                      *[torch.bfloat16] if CUDA11OrLater and SM80OrLater else []))
```

Comment:
We should be consistent. Either have the extra dtypes in `floating_types_and(...)` or on the outside. Plus, could you move `floating_types_and(...)` to the top?

Reply:
Yes, sure. Moved it and addressed the other part too. I'll keep a note about styling. Thanks